### PR TITLE
Grails 4 compatibility

### PR DESCRIPTION
**Pull request by jjelliott**
This makes the plugin work with Grails 4 by replacing a removed GORM method with a newer equivalent. 

I can tweak the dependency versions to match a different release in the 4.x.x cycle if that's required. 

I also updated the tests to use the current versions of the testing frameworks, rather than the legacy grails-test-mixins.

### CODE REVIEW

**Review by JasonTypesCodes — CHANGES_REQUESTED** [review]

File: build.gradle [hunk 1]
```diff
@@ -5,17 +5,16 @@ buildscript {
     }
     dependencies {
         classpath "org.grails:grails-gradle-plugin:$grailsVersion"
-        classpath "org.grails.plugins:hibernate5:${gormVersion-".RELEASE"}"
+		classpath "org.grails.plugins:hibernate5:7.0.5"
```
[the remark this answers is not in the document]
Comment (jjelliott) [reply build.gradle:8]: That version does not line up with the `gorm.version` variable, should I add another?

[the remark this answers is not in the document]
Comment (JasonTypesCodes) [reply build.gradle:8]: I think that is a good idea, yes

[the remark this answers is not in the document]
Comment (jjelliott) [reply build.gradle]: This has been addressed.

**Review by JasonTypesCodes — APPROVED** [review]
Nice work!

**Review by sbglasius — APPROVED** [review]
My only comment is, that you changed the TagsTaglibTest to a Spock speck, then why not the TaggableTests? 
No need to change it, just wondering 🙂

**Comment by jjelliott** [comment]
@sbglasius I converted the unit tests because something was broken, but the integration tests worked without needing to convert them. I did that a while back so I can't remember _what_ exactly was wrong, but that's why.

**Pull request merged**